traits/trait_types.py

```diff
@@ -2501,8 +2501,42 @@ def init_fast_validate(self, *args):
         """
         super().init_fast_validate(*args)
 
+        # Don't set fast validation in the no-type-check case; we need
+        # Python-level handling so that we can issue a deprecation warning
+        # for the case of validating a list.
+        # xref: enthought/traits#1626
+        if self.no_type_check:
+            return
+
         self.fast_validate = args
 
+    def validate(self, object, name, value):
+        """ Validates that the value is a valid tuple.
+        """
+        if self.no_type_check:
+            if isinstance(value, tuple):
+                return tuple(value)
```

Comment:
i'm not sure why we're calling `tuple` on a `tuple`

Reply:
Ah, good catch. I should have reverted this. It makes a difference in the case that we have an instance of a subclass of `tuple` (e.g., a named tuple of some kind), and in that case it's not 100% clear whether it's better to store a plain tuple or to store the instance of the subclass.

Either way, for this PR we should stick with the current behaviour, which is _not_ to convert an instance of a `tuple` subclass to have exact type `tuple`.
```suggestion
                return value
```

Reply:
Also worth noting that the correct behaviour is even murkier because in the case of a `Tuple` with types, we _do_ always re-create a brand new `tuple`, so we do end up converting instances of (strict) subclasses of `tuple` to plain `tuple` objects.